pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java

```diff
@@ -617,4 +1261,50 @@ public byte[][][] transformToBytesValuesMV(ValueBlock valueBlock) {
     }
     return _bytesValuesMV;
   }
+
+  @Override
+  public Pair<byte[][][], RoaringBitmap> transformToBytesValuesMVWithNull(ValueBlock valueBlock) {
+    int length = valueBlock.getNumDocs();
+    if (_bytesValuesMV == null) {
+      _bytesValuesMV = new byte[length][][];
+    }
+    RoaringBitmap bitmap;
+    DataType resultDataType = getResultMetadata().getDataType();
+    switch (resultDataType) {
+      case UNKNOWN:
+        bitmap = new RoaringBitmap();
+        bitmap.add(0L, length);
+        for (int i = 0; i < length; i++) {
+          _bytesValuesMV[i] = (byte[][]) DataSchema.ColumnDataType.BYTES_ARRAY.getNullPlaceholder();
+        }
+        break;
+      case STRING:
+        Pair<String[][], RoaringBitmap> stringResult = transformToStringValuesMVWithNull(valueBlock);
+        bitmap = stringResult.getRight();
+        ArrayCopyUtils.copy(stringResult.getLeft(), _bytesValuesMV, length);
+        break;
+      case BYTES:
+        _bytesValuesMV = transformToBytesValuesMV(valueBlock);
+        bitmap = getNullBitmap(valueBlock);
+        break;
+      default:
+        throw new IllegalStateException(String.format("Cannot read MV %s as bytes", resultDataType));
+    }
+    return ImmutablePair.of(_bytesValuesMV, bitmap);
+  }
+
+  @Override
+  public @Nullable RoaringBitmap getNullBitmap(ValueBlock valueBlock) {
+    if (_arguments == null) {
```

Comment:
`_arguments` shouldn't be `null` if all the transform functions call the `init()`. Do we plan to do that separately? If so, let's add a TODO

Reply:
yeah. added a TODO. this PR only demos the right usage of null. so not every transform func is initiated.